src/sdl/sdl_display.c

```diff
@@ -303,6 +309,71 @@ static bool sdl_resize_display(ALLEGRO_DISPLAY *display, int width, int height)
    return true;
 }
 
+static bool sdl_set_display_flag(ALLEGRO_DISPLAY *display, int flag,
+   bool flag_onoff)
+{
+   ALLEGRO_DISPLAY_SDL *sdl = (void *)display;
+   switch (flag) {
+      case ALLEGRO_FRAMELESS:
+         /* The ALLEGRO_FRAMELESS flag is backwards. */
+         SDL_SetWindowBordered(sdl->window, !flag_onoff);
+         return true;
+      case ALLEGRO_FULLSCREEN_WINDOW:
+         SDL_SetWindowFullscreen(sdl->window, flag_onoff ? SDL_WINDOW_FULLSCREEN_DESKTOP : 0);
+         return true;
+      case ALLEGRO_MAXIMIZED:
+          if (flag_onoff) {
+             SDL_MaximizeWindow(sdl->window);
+          } else {
+             SDL_RestoreWindow(sdl->window);
+          }
+         return true;
+   }
+   return false;
+}
+
+static void sdl_set_icons(ALLEGRO_DISPLAY *display, int num_icons, ALLEGRO_BITMAP *bitmaps[]) {
+   ALLEGRO_DISPLAY_SDL *sdl = (void *)display;
+   int w = al_get_bitmap_width(bitmaps[0]);
+   int h = al_get_bitmap_height(bitmaps[0]);
+   int data_size = w * h * 4;
+   (void)num_icons;
+
+   unsigned char* data = al_malloc(data_size * sizeof(data[0]));
+
+   Uint32 rmask, gmask, bmask, amask;
+#if SDL_BYTEORDER == SDL_BIG_ENDIAN
+   rmask = 0xff000000;
+   gmask = 0x00ff0000;
+   bmask = 0x0000ff00;
+   amask = 0x000000ff;
+#else // little endian, like x86
+   rmask = 0x000000ff;
+   gmask = 0x0000ff00;
+   bmask = 0x00ff0000;
+   amask = 0xff000000;
+#endif
+
+   ALLEGRO_LOCKED_REGION *lock = al_lock_bitmap(bitmaps[0], ALLEGRO_PIXEL_FORMAT_ABGR_8888, ALLEGRO_LOCK_READONLY);
+   if (lock) {
+      int i = 0;
+      for (int y = 0; y < h; y++) {
+         for (int x = 0; x < w; x++) {
+            ALLEGRO_COLOR c = al_get_pixel(bitmaps[0], x, y);
+            al_unmap_rgba(c, data+i, data+i+1, data+i+2, data+i+3);
```

Comment:
This is hardly speed critical, but you could in principle memcpy here rather than do a get/unmap.

Reply:
Correct me if I'm wrong, but I think this would have to take possibility of negative pitch into account, which Allegro can return but SDL doesn't seem to handle in SDL_CreateRGBSurfaceFrom.

Reply:
Yes, you'd have to copy row-by row, but that's still significantly faster. Again, this is just an FYI, it doesn't matter here I think.